pennylane/ops/op_math/prod.py

```diff
@@ -366,3 +366,55 @@ def simplify(self) -> Union["Prod", Sum]:
         factors = [Prod(*factor).simplify() if len(factor) > 1 else factor[0] for factor in factors]
 
         return Sum(*factors)
+
+
+def _prod_sort(op_list, wire_map: dict = None):
+    """Insertion sort algorithm that sorts a list of product factors by their wire indices, taking
+    into account the operator commutivity.
+
+    Args:
+        op_list (List[.Operator]): list of operators to be sorted
+        wire_map (dict): Dictionary containing the wire values as keys and its indexes as values.
+            Defaults to None.
+
+    Returns:
+        List[.Operator]: sorted list of operators
+    """
+
+    if isinstance(op_list, tuple):
+        op_list = list(op_list)
+
+    for i in range(1, len(op_list)):
+
+        left_op = op_list[i]
```

Comment:
So `j` is going to be less than `i`, therefore `op_list[j]` is going to be to the left of `op_list[i]`, if we think about it as `[op1, op2, op3, op4, op5]`.

Maybe `upper_op` instead?  Or is it "left" in the context of "remaining" instead of "not right"?

Reply:
In this algorithm we iterate through the operators and we always have two lists, on ordered (on the left) and another unordered (on the right). `left_op` is the operator to the left on the unordered list.

![image](https://user-images.githubusercontent.com/49232747/186850767-3fbc4e0b-e8a4-47f7-99ec-5693fb75f323.png)

Reply:
Changed the name to `key_op`